### PR TITLE
Bq public data

### PR DESCRIPTION
**Pull request by afleisc**
Modify deploy_and_run_tests so that UDFs can optionally be deployed and unit tested in bigquery-public-data.persistent_udfs dataset

### CODE REVIEW

File: udfs/tests/dataform_testing_framework/deploy_and_run_tests.sh [hunk 1]
```diff
@@ -219,6 +219,26 @@ main() {
   # for all other Dataform project directories.
   dataform install >/dev/null 2>&1
 
+  # Deploy UDFs to bigquery-public-data project
+  public_project_id="bigquery-public-data"
+  public_dataset_id="persistent_udfs"
+  if [[ -n "${_PUBLIC_DATA}"]]; then
```
Comment (danieldeleo) [inline udfs/tests/dataform_testing_framework/deploy_and_run_tests.sh:225]: everything that follows outside of this if statement should be part of an else statement. If _PUBLIC_DATA is set then we're only deploying to public data and not doing any of the other steps that come after this if statement.

Comment (danieldeleo) [inline udfs/tests/dataform_testing_framework/deploy_and_run_tests.sh:225]: Let's pass the public data project id via the env variable.

File: udfs/tests/dataform_testing_framework/deploy_and_run_tests.sh [hunk 2]
```diff
@@ -219,6 +219,26 @@ main() {
...
+      # Deploy all UDFs in the community folder
+      deploy_udfs \
+        "${public_project_id}" \
+        "${JS_BUCKET}" \
+        "community" \
+        "${public_dataset_id}${SHORT_SHA}" \
```
Comment (danieldeleo) [inline udfs/tests/dataform_testing_framework/deploy_and_run_tests.sh:231]:
```suggestion
        "${public_dataset_id}" \
```

File: udfs/tests/dataform_testing_framework/deploy_and_run_tests.sh [hunk 3]
```diff
@@ -219,6 +219,26 @@ main() {
...
+        "$(pwd)"/../../community \
+        "community_deploy"
+      # Run unit tests for all UDFs in community folder
+      test_udfs \
+        "${public_project_id}" \
+        "${public_dataset_id}${SHORT_SHA}" \
```
Comment (danieldeleo) [inline udfs/tests/dataform_testing_framework/deploy_and_run_tests.sh:237]:
```suggestion
        "${public_dataset_id}" \
```

File: udfs/tests/dataform_testing_framework/deploy_and_run_tests.sh (hunk 1 above)
Comment (danieldeleo) [reply udfs/tests/dataform_testing_framework/deploy_and_run_tests.sh:225]: alternatively, we can just add an `exit 0` after the last statement in the if block

Comment (afleisc) [reply udfs/tests/dataform_testing_framework/deploy_and_run_tests.sh]: resolved by adding if/then logic

File: udfs/tests/dataform_testing_framework/deploy_and_run_tests.sh [hunk 4]
```diff
@@ -219,63 +219,82 @@ main() {
...
-    printf "*************** "
-    printf "Testing UDFs in BigQuery dataset: %s%s" "${dataset_id}" "${SHORT_SHA}"
-    printf " ***************\n"
-
-    # SHORT_SHA environment variable below comes from
-    # cloud build when the trigger originates from a github commit.
-    if [[ $udf_dir == 'community' ]]; then
+  # Deploy UDFs to bigquery-public-data project
+  public_dataset_id="persistent_udfs"
+  if [[ -n "${_PUBLIC_PROJECT_ID}"]]; then
```
Comment (danieldeleo) [inline udfs/tests/dataform_testing_framework/deploy_and_run_tests.sh:224]:
```suggestion
  if [[ -n "${_PUBLIC_PROJECT_ID}" ]]; then
```

File: udfs/deploy.yaml [hunk 5]
```diff
@@ -25,6 +25,7 @@ steps:
     - deploy_and_run_tests.sh
   env:
     - PROJECT_ID=${PROJECT_ID}
+    - PUBLIC_PROJECT_ID=${_PUBLIC_PROJECT_ID}
```
Comment (danieldeleo) [inline udfs/deploy.yaml:28]:
```suggestion
    - PROJECT_ID=${_PROJECT_ID}
```

File: udfs/tests/dataform_testing_framework/deploy_and_run_tests.sh [hunk 6]
```diff
@@ -219,63 +219,82 @@ main() {
...
-    printf "*************** "
-    printf "Testing UDFs in BigQuery dataset: %s%s" "${dataset_id}" "${SHORT_SHA}"
-    printf " ***************\n"
-
-    # SHORT_SHA environment variable below comes from
-    # cloud build when the trigger originates from a github commit.
-    if [[ $udf_dir == 'community' ]]; then
+  # Deploy UDFs to bigquery-public-data project
+  public_dataset_id="persistent_udfs"
+  if [[ -n "${PUBLIC_PROJECT_ID}" ]]; then
```
Comment (danieldeleo) [inline udfs/tests/dataform_testing_framework/deploy_and_run_tests.sh:224]:
```suggestion
  if [[ -n "${PROJECT_ID}" && "${PROJECT_ID}" = "bigquery-public-data" ]]; then
```

File: udfs/tests/dataform_testing_framework/deploy_and_run_tests.sh [hunk 7]
```diff
@@ -219,63 +219,82 @@ main() {
...
       # Deploy all UDFs in the community folder
       deploy_udfs \
-        "${PROJECT_ID}" \
+        "${PUBLIC_PROJECT_ID}" \
```
Comment (danieldeleo) [inline udfs/tests/dataform_testing_framework/deploy_and_run_tests.sh:227]:
```suggestion
        "${PROJECT_ID}" \
```

File: udfs/tests/dataform_testing_framework/deploy_and_run_tests.sh [hunk 8]
```diff
@@ -219,63 +219,82 @@ main() {
...
-        "${udf_dir}"_deploy
+        "community" \
+        "${public_dataset_id}" \
+        "$(pwd)"/../../community \
+        "community_deploy"
       # Run unit tests for all UDFs in community folder
       test_udfs \
-        "${PROJECT_ID}" \
-        "${dataset_id}${SHORT_SHA}" \
+        "${PUBLIC_PROJECT_ID}" \
```
Comment (danieldeleo) [inline udfs/tests/dataform_testing_framework/deploy_and_run_tests.sh:235]:
```suggestion
        "${PROJECT_ID}" \
```

**Review by danieldeleo — APPROVED** [review]

**Pull request merged**